### PR TITLE
Ring buffer PR 1

### PR DESCRIPTION
**Pull request by achenchen1**

### CODE REVIEW

**Comment by achenchen1** [comment]
Looks like the first test passed - should I squash commits + copy back the modified jobs from temp into jobs2 for diffing?

**Comment by priyasiddharth** [comment]
Yes please squash commits.

File: seahorn/tmp/ring_buffer_release2/sea.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+verify_options:
+  horn-explicit-sp0: true
+  horn-bv2-tracking-mem: true
```
Comment (priyasiddharth) [inline seahorn/tmp/ring_buffer_release2/sea.yaml:3]: add newline please.

File: seahorn/jobs2/ring_buffer_acquire2/aws_ring_buffer_acquire_harness2.c [hunk 2]
```diff
@@ -67,11 +74,15 @@ int main(void) {
...
+            !sea_is_modified((char *)ring_buf.allocation) &&
+            !sea_is_modified((char *)ring_buf.allocation_end) &&
+            !sea_is_modified((char *)ring_buf.head.value) &&
+            !sea_is_modified((char *)ring_buf.tail.value));
   }
   sassert(aws_ring_buffer_is_valid(&ring_buf));
-  sassert(ring_buf.allocator == ring_buf_old.allocator);
-  sassert(ring_buf.allocation == ring_buf_old.allocation);
-  sassert(ring_buf.allocation_end == ring_buf_old.allocation_end);
+  sassert(!sea_is_modified((char *)ring_buf.allocator));
```
Comment (priyasiddharth) [inline seahorn/jobs2/ring_buffer_acquire2/aws_ring_buffer_acquire_harness2.c:84]: This is incorrect. The original check asserts that the pointer in `allocator` is unchanged. Here we are asserting that the memory pointed to by `allocator` is unchanged.

Ditto for the checks in lines 85, 86

File: seahorn/jobs2/ring_buffer_acquire2/aws_ring_buffer_acquire_harness2.c [hunk 3]
```diff
@@ -67,11 +74,15 @@ int main(void) {
     sassert(!(is_front_valid_state(&ring_buf_old) &&
               is_middle_valid_state(&ring_buf)));
   } else {
-    sassert(ring_buffers_are_equal(&ring_buf, &ring_buf_old));
+    sassert(!sea_is_modified((char *)ring_buf.allocator) &&
```
Comment (priyasiddharth) [inline seahorn/jobs2/ring_buffer_acquire2/aws_ring_buffer_acquire_harness2.c:77]: Can you provide some intuition for why `ring_buffers_are_equal` and this `sassert` are equivalent?

To me it seems like we want to check that `allocation` is unmodified and the rest of the equality checks remain.

File: seahorn/jobs2/ring_buffer_buf_belongs_to_pool2/aws_ring_buffer_buf_belongs_to_pool_harness2.c [hunk 4]
```diff
@@ -26,33 +26,25 @@ int main(void) {
...
 
   bool result = aws_ring_buffer_buf_belongs_to_pool(&ring_buf, &buf);
 
   /* assertions */
   sassert(is_member == result);
   sassert(aws_ring_buffer_is_valid(&ring_buf));
   sassert(aws_byte_buf_is_valid(&buf));
-  sassert(ring_buffers_are_equal(&ring_buf_old, &ring_buf));
-  sassert(byte_bufs_are_equal(&buf_old, &buf));
+  sassert(!sea_is_modified((char *)&ring_buf));
```
Comment (priyasiddharth) [inline seahorn/jobs2/ring_buffer_buf_belongs_to_pool2/aws_ring_buffer_buf_belongs_to_pool_harness2.c:47]: This replacement of `ring_buffers_are_equal` is different from `aws_ring_buffer_acquire_up_to_harness2.c` ... why so?

**Review by priyasiddharth — CHANGES_REQUESTED** [review]
We need to think more about how to port ring buffer checks. I've added some comments. Please get back if questions are unclear

**Comment by priyasiddharth** [comment]
This looks good to be merged. Please resolve outdated comment and fix the one new comment and I can merge it.

**Comment by priyasiddharth** [comment]
note to self:
I will merge this once tests pass

**Comment by priyasiddharth** [comment]
This CI has passed before a trivial change was made so assuming the risk and merging it in.

**Pull request merged**